### PR TITLE
Adapt options when built as sub-project

### PR DESCRIPTION
**Pull request by ClausKlein**
skipp tests, example, and install targets

this fix https://github.com/taocpp/json/issues/94

### CODE REVIEW

**Comment by d-frey** [comment]
OK, I'll accept this PR, but expect some cleanup commits from me afterwards.

**Comment by d-frey** [comment]
If you like (and you think it is useful), similar PRs (with adapted variable names) for the PEGTL and taoPQ would be appreciated.

**Review by ClausKlein — COMMENTED** [review]
> If you like (and you think it is useful), similar PRs (with adapted variable names) for the PEGTL and taoPQ would be appreciated.

Is PEGTL_USE_BOOST_FILESYSTEM still needed on some build systems? 
If true, a PR may needed.

**Comment by d-frey** [comment]
It is not needed because of any build system, it is only available as a choice for users that simply want to use it instead of std::…, maybe because they are using boost in there project anyways.

File: CMakeLists.txt [hunk 1]
```diff
@@ -1,65 +1,90 @@
...
 # examples
-option(TAOCPP_JSON_BUILD_EXAMPLES "Build example programs" ON)
+option(TAOCPP_JSON_BUILD_EXAMPLES "Build example programs" ${MASTER_PROJECT})
 if(TAOCPP_JSON_BUILD_EXAMPLES)
   add_subdirectory(src/example/json)
 endif()
 
-# Make package findable
-configure_file(.cmake/taocpp-json-config.cmake.in taocpp-json-config.cmake @ONLY)
+option(TAOCPP_JSON_INSTALL "Generate the install target" ${MASTER_PROJECT})
```
[the remark this answers is not in the document]
Comment (d-frey) [reply CMakeLists.txt:59]: AFAIK creating the install target doesn't do anything on its own, the user still needs to actually ask for the project to be installed, right? Or does the above (our current code) always create some configure file? In that case, a PR might be necessary.

[the remark this answers is not in the document]
Comment (ClausKlein) [reply CMakeLists.txt:59]: Yes,  the install target is not the default target.

**Pull request merged**